finetune/util/table_labeler.py

```diff
@@ -0,0 +1,331 @@
+"""
+Finetune-style interface for running a pipeline of table and non-table models.
+"""
+import copy
+import functools
+import typing as t
+
+from finetune.base_models import TableRoBERTa, RoBERTa
+from finetune.util.metrics import sequences_overlap
+from finetune.scheduler import Scheduler
+
+from finetune import SequenceLabeler
+
+
+class TableETL:
+    def __init__(
+        self,
+        drop_table_from_text_labels: bool = False,
+        drop_table_from_text_preds: bool = True,
+    ):
+        self.drop_table_from_text_labels = drop_table_from_text_labels
+        self.drop_table_from_text_preds = drop_table_from_text_preds
+
+    def get_table_text(self, text: str, doc_offsets: t.List[t.Dict[str, int]]) -> str:
+        return "\n".join(text[c["start"] : c["end"]] for c in doc_offsets)
+
+    def fix_spans(
+        self,
+        spans: t.List[t.Dict[str, t.Any]],
+        chunk: t.List[t.Dict[str, t.Dict[str, int]]],
+        output_space_text: str,
+        input_space: str,
+        output_space: str,
+        strict: bool = False,
+    ) -> t.List[t.Dict[str, t.Any]]:
+        """
+        convers a list of spans (anything with a start, end and optionally text, usually labels or context)
+        between 2 character spaces (document and table)
+
+        Will do 1 -> many splits when a span spreads across multiple chunks.
+
+        Args:
+           spans: The spans (labels or context) to move between the character spaces.
+           chunk: The chunks a list of dicts including input_space and output_space keys.
+           output_space_text: The text of the output space, used for validation.
+           input_space: a key into chunks of the space the current spans are in.
+           output_space: a key into chunks of the space to move the current spans
+           strict: bool, whether to assert that every span is matched by at least one chunk.
+
+        """
+        spans_out = []
+        for orig_span in spans:
+            matched_chunk = False
+            for sub_chunk in chunk:
+                if sequences_overlap(orig_span, sub_chunk[input_space]):
```

Comment:
Could early exit by checking if sequence_overlap and matched_chunk is already set to True